drivers/i2c/i2c_sc18is600.c

```diff
@@ -0,0 +1,315 @@
+{
+	size_t bufsize = 5 + count_w;
+	u8_t buf[bufsize];
+	size_t index = 0;
+
+	buf[index++] = READ_AFTER_WRITE_DEVICE;
+	buf[index++] = count_w;
+	buf[index++] = count_r;
+	buf[index++] = addr_w;
+	for (int i = 0; i < count_w; i++)
+		buf[index++] = data_w[i];
+	buf[index++] = addr_r;
+	i2c_sc18is600_spi_send_buffer(dev, buf, bufsize);
+}
+
+void i2c_sc18is600_read_from_device(struct device *dev, u8_t addr,
+		u8_t count, u8_t *data)
+{
+	u8_t buf[3] = {READ_N_BYTES_FROM_DEVICE, count, addr};
+
+	i2c_sc18is600_spi_send_buffer(dev, buf, 3);
+}
+
+void i2c_sc18is600_read_from_bridge_rx_buffer(struct device *dev,
+		u8_t count, u8_t *data)
+{
+	size_t bufsize = 1 + count;
+	u8_t tx_buffer[bufsize];
+	u8_t rx_buffer[bufsize];
+	size_t index = 0;
+
+	tx_buffer[index++] = READ_BRIDGE_RX_BUFFER;
+
+	for (int i = 0; i < count; i++) {
+		tx_buffer[index++] = 0xFF; /* Send dummy bytes */
+	}
+	/* Set delay between two consecutive spi frames
+	 * to 7 clock cycles, SC18IS600 specific
+	 */
+	SPI1_REG(SPI_REG_DINTERXFR) = 0x07;
+	i2c_sc18is600_spi_transceive_buffer(dev, tx_buffer, bufsize,
+			rx_buffer, bufsize);
+	SPI1_REG(SPI_REG_DINTERXFR) = 0x00; /* Remove delay */
+
+	for (u8_t i = 0; i < count; i++) {
+		data[i] = rx_buffer[i + 1];
+	}
+}
+
+void i2c_sc18is600_write_to_device(struct device *dev, u8_t addr,
+		u8_t count, u8_t *data)
+{
+	size_t bufsize = 3 + count;
+	size_t index = 0;
+	u8_t tx_buffer[bufsize];
+
+	tx_buffer[index++] = WRITE_N_BYTES_TO_DEVICE;
+	tx_buffer[index++] = count;
+	tx_buffer[index++] = addr;
+	for (int i = 0; i < count; i++)
+		tx_buffer[index++] = data[i];
+	i2c_sc18is600_spi_send_buffer(dev, tx_buffer, bufsize);
+}
+
+/* INITIALIZATION */
+static int i2c_sc18is600_init(struct device *dev)
+{
+	struct i2c_sc18is600_runtime *data =
+		(struct i2c_sc18is600_runtime *) dev->driver_data;
+
+	data->spi_d = device_get_binding(CONFIG_SPI_0_NAME);
+	if (!data->spi_d)
+		printk("spi device binding inside i2c driver failed\n");
+	return 0;
+}
+
+/* POWER MANAGEMENT */
+
+void i2c_sc18is600_remove_pinmux(void)
+{
+	GPIO_REG(GPIO_IOF_EN) &= ~((1 << 2) | (1 << 3)
+			| (1 << 4) | (1 << 5)
+			| (1 << 9) | (1 << 10));
+	GPIO_REG(GPIO_INPUT_EN) &= ~((1 << 2) | (1 << 3)
+			| (1 << 4) | (1 << 5)
+			| (1 << 9) | (1 << 10));
+	GPIO_REG(GPIO_OUTPUT_EN) &= ~((1 << 2) | (1 << 3)
+			| (1 << 4) | (1 << 5)
+			| (1 << 9) | (1 << 10));
+}
+
+/* API */
+static int i2c_sc18is600_configure(struct device *dev, u32_t dev_config)
+{
+	struct i2c_sc18is600_runtime *data =
+		(struct i2c_sc18is600_runtime *) dev->driver_data;
+	i2c_sc18is600_spi_setup(dev);
+	data->i2c_config = dev_config;
+	i2c_sc18is600_write_to_bridge_register(dev, 0x02, 0x05);
+	return 0;
+};
+
+static int i2c_sc18is600_transfer(struct device *dev, struct i2c_msg *msgs,
+		u8_t num_msgs, u16_t addr)
+{
+	struct i2c_sc18is600_runtime *data =
+		(struct i2c_sc18is600_runtime *) dev->driver_data;
+	i2c_sc18is600_configure(dev, data->i2c_config);
+
+	if (num_msgs > 0) {
```

Comment:
It's a bit ugly to have all these if imbricated, you end up having not nice indentation.
Just change first test: if (num_msgs != 2) { return 0; }